### PR TITLE
feat: support platform override.

### PR DESCRIPTION
**Pull request by frezbo**
feat: support platform override.
    
Support setting `platform` for dependencies and also allow overriding `platform` for a specific package.
    
Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: internal/pkg/convert/graph.go [hunk 1]
```diff
@@ -29,19 +30,24 @@ type GraphLLB struct {
 	LocalContext llb.State
 
 	baseImageProcessor llbProcessor
-	cache              map[*solver.PackageNode]llb.State
+	cache              map[cacheState]llb.State
 
 	commonRunOptions []llb.RunOption
 }
 
+type cacheState struct {
```
Comment (smira) [inline internal/pkg/convert/graph.go:38]:
```suggestion
type cacheKey struct {
```

File: internal/pkg/convert/graph.go [hunk 2]
```diff
@@ -55,6 +61,23 @@ func NewGraphLLB(graph *solver.PackageGraph, options *environment.Options) *Grap
 	return result
 }
 
+func (graph *GraphLLB) convertPlatform(platform string) *v1.Platform {
```
Comment (smira) [inline internal/pkg/convert/graph.go:64]: this seems to be not related to the graph, probably move it to be a function on its own (to a separate file)?

File: internal/pkg/convert/node.go [hunk 3]
```diff
@@ -327,8 +348,22 @@ func (node *NodeLLB) finalize(root llb.State) llb.State {
...
+		PackageNode: node.PackageNode,
+	}
+
+	if platform != "" {
+		v1Platform, err := node.convertPlatform(platform)
+		if err != nil {
+			return llb.Scratch(), err
+		}
+
+		node.Platform = Platform{v1Platform}
```
Comment (smira) [inline internal/pkg/convert/node.go:362]: this looks strange that it sets it on the node, does it make more sense to pass platform to `NewNodeLLB` ?

Comment (frezbo) [reply internal/pkg/convert/node.go]: fixed

**Review by smira — APPROVED** [review]
